crates/sui-core/src/transaction_driver/mod.rs

```diff
@@ -193,9 +193,13 @@ where
 
 // Chooses the percentage of transactions to be driven by TransactionDriver.
 pub fn choose_transaction_driver_percentage() -> u8 {
-    // Currently, TD cannot work in non-test environments.
-    if std::env::var(sui_types::digests::SUI_PROTOCOL_CONFIG_CHAIN_OVERRIDE_ENV_VAR_NAME).is_ok() {
-        return 0;
+    // Currently, TD cannot work in mainnet.
+    if let Ok(chain) =
+        std::env::var(sui_types::digests::SUI_PROTOCOL_CONFIG_CHAIN_OVERRIDE_ENV_VAR_NAME)
+    {
+        if chain == "mainnet" {
+            return 0;
+        }
     }
```

Comment:
What if someone doesn't set the `SUI_PROTOCOL_CONFIG_CHAIN_OVERRIDE_ENV_VAR_NAME` at all but sets the `TRANSACTION_DRIVER` ? Wouldn't that enable it?